### PR TITLE
Dataflow Worker: Add worker_id to windmill request headers

### PR DESCRIPTION
**Pull request by rangadi**
Adds 'worker_id' to header for 'commitWork' and 'getData' streaming requests from Dataflow worker to windmill. 

+R: @boyuanzz

### CODE REVIEW

**Comment by boyuanzz** [comment]
Hey @kennknowles, could you please take a look at this PR?

**Comment by boyuanzz** [comment]
Run Java PreCommit

**Comment by boyuanzz** [comment]
seems like the new added worker field broke 3 unit tests: https://builds.apache.org/job/beam_PreCommit_Java_Phrase/321/testReport/. Could you please adjust the failed unit tests in this PR? @rangadi

**Comment by rangadi** [comment]
Thanks @boyuanzz. I have updated the affected tests. `./gradlew test -p runners/google-cloud-dataflow-java/worker` succeeded on my machine.

**Comment by boyuanzz** [comment]
@lukecwik could you please help merge this PR after all tests passed?

**Comment by boyuanzz** [comment]
Run Java PreCommit

**Comment by boyuanzz** [comment]
Run Java PreCommit

**Comment by boyuanzz** [comment]
This PR is ready to merge if there is no more concern. Could any committer help merge this PR?

File: runners/google-cloud-dataflow-java/worker/windmill/src/main/proto/windmill.proto [hunk 1]
```diff
@@ -446,6 +446,8 @@ message StreamingCommitWorkRequest {
 message JobHeader {
   optional string job_id = 1;
   optional string project_id = 2;
+  // Worker id is meant for logging only. Do not rely on it for other decisions.
```
Comment (lukecwik) [inline runners/google-cloud-dataflow-java/worker/windmill/src/main/proto/windmill.proto]: Hah. 5 mins later, being used for some mission critical thing.

Comment (rangadi) [reply runners/google-cloud-dataflow-java/worker/windmill/src/main/proto/windmill.proto]: Hmm... in a design discussion?

**Pull request merged**